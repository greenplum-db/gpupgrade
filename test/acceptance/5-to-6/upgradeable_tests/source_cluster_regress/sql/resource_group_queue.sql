-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test to ensure that resource groups and resource queues can be upgraded.

--------------------------------------------------------------------------------
-- Create and setup upgradeable objects
--------------------------------------------------------------------------------

-- Ensure overridden attributes of the default queue and default groups are upgraded
ALTER RESOURCE QUEUE pg_default WITH (ACTIVE_STATEMENTS=2, MIN_COST=1700, MAX_COST=2000, COST_OVERCOMMIT=false, PRIORITY=MIN, MEMORY_LIMIT ='10MB');

ALTER RESOURCE GROUP admin_group SET CONCURRENCY 5;
ALTER RESOURCE GROUP admin_group SET CPU_RATE_LIMIT 5;
ALTER RESOURCE GROUP admin_group SET MEMORY_LIMIT 5;
ALTER RESOURCE GROUP admin_group SET MEMORY_SHARED_QUOTA 5;
ALTER RESOURCE GROUP admin_group SET MEMORY_SPILL_RATIO 5;

ALTER RESOURCE GROUP default_group SET MEMORY_LIMIT 5;
ALTER RESOURCE GROUP default_group SET CPU_RATE_LIMIT 5;
ALTER RESOURCE GROUP default_group SET MEMORY_LIMIT 5;
ALTER RESOURCE GROUP default_group SET MEMORY_SHARED_QUOTA 5;
ALTER RESOURCE GROUP default_group SET MEMORY_SPILL_RATIO 5;

-- Validate attributes of resource queues
SELECT rsqname, rsqcountlimit, rsqcostlimit, rsqovercommit,
       rsqignorecostlimit, resname, ressetting
    FROM pg_resqueue r, pg_resqueuecapability c, pg_resourcetype t
    WHERE r.oid=c.resqueueid AND c.restypid=t.restypid
    ORDER BY rsqname;

-- Validate attributes of resource groups
SELECT groupname, concurrency, proposed_concurrency, cpu_rate_limit,
       memory_limit, proposed_memory_limit, memory_shared_quota,
       proposed_memory_shared_quota, memory_spill_ratio, proposed_memory_spill_ratio
    FROM gp_toolkit.gp_resgroup_config
    ORDER BY groupname;

-- Validate resource queue and group assignment to test_role
SELECT rolname, rsqname, rsgname FROM pg_roles, pg_resgroup, pg_resqueue
    WHERE pg_roles.rolresgroup=pg_resgroup.oid
    AND pg_roles.rolresqueue=pg_resqueue.oid
    AND rolname='test_role';
