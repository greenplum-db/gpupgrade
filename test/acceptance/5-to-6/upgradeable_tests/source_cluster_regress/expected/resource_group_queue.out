-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test to ensure that resource groups and resource queues can be upgraded.

--------------------------------------------------------------------------------
-- Create and setup upgradeable objects
--------------------------------------------------------------------------------
CREATE RESOURCE QUEUE test_queue WITH (ACTIVE_STATEMENTS=2, MIN_COST=1700, MAX_COST=2000, COST_OVERCOMMIT=false, PRIORITY=MIN, MEMORY_LIMIT ='10MB');
CREATE

CREATE RESOURCE GROUP test_group WITH (CONCURRENCY=5, CPU_RATE_LIMIT=5, MEMORY_LIMIT=5, MEMORY_SHARED_QUOTA=5, MEMORY_SPILL_RATIO=5);
CREATE

CREATE ROLE test_role resource group test_group resource queue test_queue;
CREATE

-- Ensure overridden attributes of the default queue and default groups are upgraded
ALTER RESOURCE QUEUE pg_default WITH (ACTIVE_STATEMENTS=2, MIN_COST=1700, MAX_COST=2000, COST_OVERCOMMIT=false, PRIORITY=MIN, MEMORY_LIMIT ='10MB');
ALTER

ALTER RESOURCE GROUP admin_group SET CONCURRENCY 5;
ALTER
ALTER RESOURCE GROUP admin_group SET CPU_RATE_LIMIT 5;
ALTER
ALTER RESOURCE GROUP admin_group SET MEMORY_LIMIT 5;
ALTER
ALTER RESOURCE GROUP admin_group SET MEMORY_SHARED_QUOTA 5;
ALTER
ALTER RESOURCE GROUP admin_group SET MEMORY_SPILL_RATIO 5;
ALTER

ALTER RESOURCE GROUP default_group SET MEMORY_LIMIT 5;
ALTER
ALTER RESOURCE GROUP default_group SET CPU_RATE_LIMIT 5;
ALTER
ALTER RESOURCE GROUP default_group SET MEMORY_LIMIT 5;
ALTER
ALTER RESOURCE GROUP default_group SET MEMORY_SHARED_QUOTA 5;
ALTER
ALTER RESOURCE GROUP default_group SET MEMORY_SPILL_RATIO 5;
ALTER

-- Validate attributes of resource queues
SELECT rsqname, rsqcountlimit, rsqcostlimit, rsqovercommit, rsqignorecostlimit, resname, ressetting FROM pg_resqueue r, pg_resqueuecapability c, pg_resourcetype t WHERE r.oid=c.resqueueid AND c.restypid=t.restypid ORDER BY rsqname;
 rsqname    | rsqcountlimit | rsqcostlimit | rsqovercommit | rsqignorecostlimit | resname      | ressetting 
------------+---------------+--------------+---------------+--------------------+--------------+------------
 pg_default | 2.0           | 2000.0       | f             | 1700.0             | priority     | min        
 pg_default | 2.0           | 2000.0       | f             | 1700.0             | memory_limit | 10MB       
 test_queue | 2.0           | 2000.0       | f             | 1700.0             | memory_limit | 10MB       
 test_queue | 2.0           | 2000.0       | f             | 1700.0             | priority     | min        
(4 rows)

-- Validate attributes of resource groups
SELECT groupname, concurrency, proposed_concurrency, cpu_rate_limit, memory_limit, proposed_memory_limit, memory_shared_quota, proposed_memory_shared_quota, memory_spill_ratio, proposed_memory_spill_ratio FROM gp_toolkit.gp_resgroup_config ORDER BY groupname;
 groupname     | concurrency | proposed_concurrency | cpu_rate_limit | memory_limit | proposed_memory_limit | memory_shared_quota | proposed_memory_shared_quota | memory_spill_ratio | proposed_memory_spill_ratio 
---------------+-------------+----------------------+----------------+--------------+-----------------------+---------------------+------------------------------+--------------------+-----------------------------
 admin_group   | 5           | 5                    | 5              | 5            | 5                     | 5                   | 5                            | 5                  | 5                           
 default_group | 20          | 20                   | 5              | 5            | 5                     | 5                   | 5                            | 5                  | 5                           
 test_group    | 5           | 5                    | 5              | 5            | 5                     | 5                   | 5                            | 5                  | 5                           
(3 rows)

-- Validate resource queue and group assignment to test_role
SELECT rolname, rsqname, rsgname FROM pg_roles, pg_resgroup, pg_resqueue WHERE pg_roles.rolresgroup=pg_resgroup.oid AND pg_roles.rolresqueue=pg_resqueue.oid AND rolname='test_role';
 rolname   | rsqname    | rsgname    
-----------+------------+------------
 test_role | test_queue | test_group 
(1 row)
